File: src/boiler/erlang/day3.erl
-module(day3).
-export([day_three/0]).


print_weird() ->
    io:fwrite("Weird~n", []).

print_notweird() ->
    io:fwrite("Not weird~n", []).

even(X) when X >= 6, X =< 20 ->
    print_weird();
even(_) ->
    print_notweird().
    

odd() ->
    print_weird().

eval_num(X) ->
    if (X rem 2) ==1 ->
        odd();
    true ->
        even(X)
    end.


day_three() ->
    {ok, [Num]} = io:fread("", "~d"),
    eval_num(Num).

File: src/boiler/elixir/day3.exs
defmodule Day3 do
    def print_weird() do
        IO.puts("Weird")
    end
    def print_notweird() do
        IO.puts("Not weird")
    end
    def even(num) when 
        is_integer(num) and num > 6 and num < 20 do
        print_weird()
    end
    def even(_) do
        print_notweird()
    end
    def odd() do
        print_weird()
    end
    def weird_test(num) do
        if ( rem(num, 2) == 0) do
            even(num)
        else
            odd()
        end
    end
    def day3 do
        num = String.to_integer(String.trim(IO.gets("Enter number ")))
        weird_test(num)
    end
end
